processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -470,20 +480,71 @@ public String apply(@Nullable String input)
       }
     };
 
-    return makeIndexFiles(indexes, outDir, progress, mergedDimensions, mergedMetrics, null, rowMergerFn, indexSpec);
+    return makeIndexFiles(
+        indexes,
+        mergeAggregators(indexes),
+        outDir,
+        progress,
+        mergedDimensions,
+        mergedMetrics,
+        null,
+        rowMergerFn,
+        indexSpec
+    );
+  }
+
+  public static final AggregatorFactory[] mergeAggregators(List<IndexableAdapter> indexes)
```

Comment:
can we add some javadocs to clarify what the expected behavior is when merging indices? i.e. there are a few things I think we should explicitly state, e.g. what happens if there is a naming conflict, how does the order of the list affect which aggregators are kept? Also, in some cases the aggregators may not be compatible, e.g. if a name conflict redefines a longSum metric as a complex metric, it will probably break unexpectedly somewhere downstream.